code/modules/projectiles/bullet.dm

```diff
@@ -453,6 +453,7 @@ toxic - poisons
 
 	on_hit(atom/hit, direction, obj/projectile/P)
```

Comment:
shouldn't the parent return if it hit or not? or, why is this called when not hitting something?

Reply:
Space bees have a special Cross() proc that lets them get hit by projectiles despite having zero density.

Reply:
yes, and? 
![image](https://user-images.githubusercontent.com/4741640/224106694-ac4b1438-3032-4f95-ba2b-a7b630bcb9c4.png)

Reply:
![image](https://user-images.githubusercontent.com/58530985/224562261-4e031f73-66f6-4d28-a469-988f6fdb32d5.png)
And if a projectile passes through an object that returns FALSE for Cross() but is not actually dense, it will hit that object and continue moving without deleting itself. This means the dart projectile was calling on_hit() multiple times, and thus dropping more than one dart.

Reply:
Yeah I personally think piercing objcritters without it being specific ammo designed to pierce is definitely not intended.
I just think the whole on hit multiple times is definitely very smelly.